sansio_lsp_client/io_handler.py

```diff
@@ -149,9 +149,14 @@ def _parse_one_message(
 
     def parse_request_or_response(data: JSONDict,) -> t.Union[Request, Response]:
         del data["jsonrpc"]
-        return parse_obj_as(t.Union[Request, Response], data)  # type: ignore
+        # MY Workaround: pydantic parses optional union(list,dict) as dict first
+        if isinstance(data.get('result'), list):
+            return parse_obj_as(ResponseList, data)  # type: ignore
```

Comment:
I don't think `# type: ignore` is quite right here. This function should return `Union[Request, Response]`, not `ResponseList`.